application/backend/app/services/datumaro_converter.py

```diff
@@ -173,7 +198,7 @@ def convert_classification_dataset(
     def _convert_sample(
         dataset_item: DatasetItem, image_path: str, project_labels_ids: list[UUID]
     ) -> ClassificationSample | None:
-        if dataset_item.annotation_data is None:
+        if dataset_item.annotation_data is None or dataset_item.subset is None:
             return None
```

Comment:
Skipping an item if it doesn't have a subset is unexpected behavior for this function; its responsibility should be limited to conversion, not filtering. For the training usecase, unassigned items should be filtered out elsewhere, by calling `DatasetService.get_dm_dataset(task=project.task, annotation_status=DatasetItemAnnotationStatus.REVIEWED)`.